modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -206,6 +212,60 @@ export default function AuditBarChart({rtl, vocabularies}) {
 
 	const [tooltip, setTooltip] = useState(null);
 
+	const onBarClick = (assetCategoryIds) => {
+		if (assetCategoryIds.length) {
+			let uri = window.location.href;
+
+			if (!uri.includes('resetBarsCategoryFiltersURL')) {
+				uri = Liferay.Util.addParams(
+					'resetBarsCategoryFiltersURL=' + encodeURIComponent(uri),
+					uri
+				);
+			}
+
+			const params = new URLSearchParams(window.location.search);
+
+			const categories = params.getAll(namespace + 'assetCategoryId');
+
+			if (categories.length) {
+				categories.map((category) => {
+					uri = uri.replace(
+						namespace + 'assetCategoryId=' + category,
+						''
+					);
+				});
+			}
+
+			assetCategoryIds.map((assetCategoryId) => {
+				if (assetCategoryId !== 'none') {
+					uri = Liferay.Util.addParams(
+						namespace + 'assetCategoryId=' + assetCategoryId,
+						uri
+					);
+				}
+			});
+
+			Liferay.Util.navigate(uri);
+		}
+	};
+
+	const handleKeydown = useCallback((event) => {
+		const uri = window.location.href;
+
+		if (
+			ESCAPE_KEYS.includes(event.key) &&
+			uri.includes('resetBarsCategoryFiltersURL')
+		) {
+			const params = new URLSearchParams(uri);
+
+			Liferay.Util.navigate(
+				decodeURIComponent(params.get('resetBarsCategoryFiltersURL'))
+			);
+		}
+	}, []);
```

Comment:
Since this callback isn't dependent on any props in this component, you can hoist this outside of the component and get rid of the `useCallback`.


```js
const handleKeydown = (event) => {
	const uri = window.location.href;

	if (
		ESCAPE_KEYS.includes(event.key) &&
		uri.includes('resetBarsCategoryFiltersURL')
	) {
		const params = new URLSearchParams(uri);

		Liferay.Util.navigate(
			decodeURIComponent(params.get('resetBarsCategoryFiltersURL'))
		);
	}
};
```

Reply:
Of course! I'll move this method outside the component 😉